sdcm/nemesis.py

```diff
@@ -1337,56 +1333,61 @@ def _check_pod_stats(node):
-        for node_terminate_method_name in self._get_kubernetes_node_break_methods():
-            self.set_target_node(rack=random.choice(list(self.cluster.racks)),
-                                 allow_only_last_node_in_rack=True)
-            InfoEvent(
-                f'OperatorNodeTerminateDecommissionAdd ({node_terminate_method_name}) {self.target_node}').publish()
-            self._disrupt_terminate_decommission_add_node_kubernetes(self.target_node, node_terminate_method_name)
-
-    def disrupt_replace_node_kubernetes(self):
+    def _disrupt_kubernetes_then_replace_scylla_node(self, disruption_method):
         if not self._is_it_on_kubernetes():
-            raise UnsupportedNemesis('OperatorNodeReplace is supported only on kubernetes')
-        old_uid = self.target_node.k8s_pod_uid
-        self.log.info('TerminateNode %s', self.target_node)
-        self.log.info('Mark %s to be replaced', self.target_node)
-        self.target_node.wait_for_svc()
-        self.target_node.mark_to_be_replaced()
-        self._kubernetes_wait_till_node_up_after_been_recreated(self.target_node, old_uid=old_uid)
-
-    def _disrupt_terminate_decommission_add_node_kubernetes(self, node, node_terminate_method_name):  # pylint: disable=invalid-name
-        self.log.info('Terminate %s', node)
-        node_terminate_method = getattr(node, node_terminate_method_name)
-        node_terminate_method()
-        self.log.info('Decommission %s', node)
-        self.cluster.decommission(node, soft_timeout=MAX_TIME_WAIT_FOR_DECOMMISSION)
-        new_node = self.add_new_node(rack=node.rack)
-        self.unset_current_running_nemesis(new_node)
-
-    def _disrupt_terminate_and_replace_node_kubernetes(self, node, node_terminate_method_name):  # pylint: disable=invalid-name
+            raise UnsupportedNemesis('Supported only on kubernetes')
+        self.set_target_node()
+        node = self.target_node
+        InfoEvent(f'Running {disruption_method} on K8S node that hosts {node} scylla pod').publish()
         old_uid = node.k8s_pod_uid
         self.log.info('TerminateNode %s (uid=%s)', node, old_uid)
-        node_terminate_method = getattr(node, node_terminate_method_name)
-        node_terminate_method()
+        getattr(node, disruption_method)()
         node.wait_till_k8s_pod_get_uid(ignore_uid=old_uid)
         old_uid = node.k8s_pod_uid
         self.log.info('Mark %s (uid=%s) to be replaced', node, old_uid)
         node.wait_for_svc()
         node.mark_to_be_replaced()
         self._kubernetes_wait_till_node_up_after_been_recreated(node, old_uid=old_uid)
 
+    def disrupt_drain_kubernetes_node_then_decommission_and_add_scylla_node(self):
+        self._disrupt_kubernetes_then_decommission_and_add_scylla_node('drain_k8s_node')
+
+    def disrupt_terminate_kubernetes_host_then_decommission_and_add_scylla_node(self):
+        # NOTE: affected by the https://github.com/scylladb/scylla-operator/issues/643
```

Comment:
Is this note still relevant? That issue still reproduce?

Reply:
Whole change is actual. The duplicated PVs bug is still existing, but it is harmless as I can see. THe decommission nemesis has it too, but passes ok.